platform/pulp/platform/models/repository.py

```diff
@@ -1,10 +1,329 @@
+    description = models.TextField(blank=True, default='')
+
+    last_content_added = models.DateTimeField(blank=True, null=True)
+    last_content_removed = models.DateTimeField(blank=True, null=True)
+
+    scratchpad = fields.GenericRelation(Scratchpad)
+    notes = fields.GenericRelation(Notes)
+
+    content = models.ManyToManyField('Content', through='RepositoryContent')
+
+    @property
+    def content_summary(self):
+        """
+        The contained content summary.
+
+        :return: A dict of {<type>: <count>}
+        :rtype:  dict
+        """
+        mapping = self.content.values('type').annotate(count=models.Count('type'))
+        return {m['type']: m['count'] for m in mapping}
+
+
+class RepositoryGroup(Model):
+    """
+    A group of repositories.
+
+    Fields:
+
+    :cvar name: The group name.
+    :type name: models.TextField
+
+    :cvar: description: An optional description.
+    :type: models.TextField
+
+    Relations:
+    :cvar notes: Arbitrary group properties.
+    :type notes: fields.GenericRelation
+
+    :cvar members: Repositories associated with the group.
+    :type members: models.ManyToManyField
+    """
+    name = models.TextField(db_index=True, unique=True)
+    description = models.TextField(blank=True, default='')
+
+    members = models.ManyToManyField('Repository')
+    scratchpad = fields.GenericRelation(Scratchpad)
+    notes = fields.GenericRelation(Notes)
+
+
+class Plugin(MasterModel):
+    """
+    An Abstract model for plugins.
+
+    Fields:
+
+    :cvar name: The plugin name.
+    :type type: models.TextField
+
+    :cvar type: The plugin type.
+    :type type: models.TextField
+
+    :cvar last_updated: When the plugin was last updated.
+    :type last_updated: fields.DateTimeField
+
+    Relations:
+
+    """
+    name = models.TextField(db_index=True)
+    type = models.TextField()
+    last_updated = models.DateTimeField(auto_now=True, blank=True, null=True)
+
+    class Meta:
+        abstract = True
+
+
+class Importer(Plugin):
+    """
+    A abstract content importer.
+
+    Fields:
+
+    :cvar feed_url: The URL of an external content source.
+    :type feed_url: models.TextField
+
+    :cvar validate: Validate the imported context.
+    :type validate: models.BooleanField
+
+    :cvar ssl_ca_certificate: A PEM encoded CA certificate used for authentication.
+    :type ssl_ca_certificate: models.TextField
+
+    :cvar ssl_client_certificate: A PEM encoded client certificate used for authentication.
+    :type ssl_client_certificate: models.TextField
+
+    :cvar ssl_client_key: A PEM encoded private key used for authentication.
+    :type ssl_client_key: models.TextField
+
+    :cvar ssl_validation: Indicates whether SSL peer validation must be performed.
+    :type ssl_validation: models.BooleanField
+
+    :cvar proxy_url: The optional proxy URL. Format: scheme://user:password@host:port
+    :type proxy_url: models.ForeignKey
```

Comment:
The `proxy_url` consolidates the removal of the individual `proxy_` fields.  The will necessitate a (good) change  in Nectar which currently builds a URL using the parts.  There is no value in storing the URL components instead of the URL.